### PR TITLE
Support Ingress and Ingress class v1

### PR DESCRIPTION
**Pull request by pleshakov**
### Proposed changes
- Use Ingress v1 in the code
- Reject Ingresses with resource backends, because the IC doesn't support such backends:
```yaml
          - path: /icons
            pathType: ImplementationSpecific
            backend:
              resource:
                apiGroup: k8s.example.com
                kind: StorageBucket
                name: icon-assets
```
- Use IngressClass v1 in the code
- Remove k8s < 1.19 from nightly

Note: the existing tests work for k8s 1.19-1.21
in a separate PR we will update the tests so that they work for k8s 1.19-1.22

The PR doesn't change the minimum supported k8s version  for the IC - that will be done separately.
The PR doesn't update any docs - that will be done separately.

### CODE REVIEW

**Comment by pleshakov** [comment]
(forced pushed to remove linting errors)

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
     needs: [build, binary, unit-tests]
     strategy:
       matrix:
-        k8s: [1.21.1, 1.20.7, 1.19.11, 1.18.19, 1.17.17, 1.16.15]
+        k8s: [1.21.1, 1.20.7, 1.19.11]
```
Comment (lucacome) [inline .github/workflows/nightly.yml:124]:
```suggestion
        k8s: [1.21.2, 1.20.7, 1.19.11]
```
new minor version of 1.21

**Review by lucacome — APPROVED** [review]

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (vepatel) [reply .github/workflows/nightly.yml:124]: no 1.22?

**Review by soneillf5 — APPROVED** [review]

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (pleshakov) [reply .github/workflows/nightly.yml:124]: @lucacome I'll send a separate PR. it will also change the version in the edge version

Comment (pleshakov) [reply .github/workflows/nightly.yml:124]: @vepatel yep, no 1.22 for now

**Pull request merged**